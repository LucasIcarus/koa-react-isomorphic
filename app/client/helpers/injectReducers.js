/* @flow */
import { combineReducers } from "redux-immutable";

export default (store: Object, reducers: Object) => {
  store.reducers = {
    ...store.reducers,
    ...reducers
  };

  store.replaceReducer(combineReducers(store.reducers));
};
