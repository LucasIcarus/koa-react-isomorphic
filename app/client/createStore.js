/* @flow */
/* global process */
import { fromJS } from "immutable";
import { combineReducers } from "redux-immutable";
import { applyMiddleware, compose, createStore } from "redux";
import thunkMiddleware from "redux-thunk";
import { createLogger } from "redux-logger";
import injectReducers from "./helpers/injectReducers";
import reducers from "./createReducer";

const middlewares = [thunkMiddleware];
const enhancers = [];

// support for development
if (process.env.RUNTIME_ENV === "client") {
  if (process.env.NODE_ENV === "development") {
    middlewares.push(createLogger({
      level: "info",
      stateTransformer: state => state.toJS()
    }));
  }

  if (window.__REDUX_DEVTOOLS_EXTENSION__) {
    enhancers.push(global.__REDUX_DEVTOOLS_EXTENSION__());
  }
}

export default (initialState: Object = {}) => {
  const store = createStore(
    combineReducers(reducers),
    fromJS(initialState),
    compose(applyMiddleware(...middlewares), ...enhancers)
  );

  // enable async reducers for each page load
  // $FlowFixMe
  store.reducers = reducers;

  if (process.env.NODE_ENV === "development" && module.hot) {
    // $FlowFixMe
    module.hot.accept("./createReducer", () =>
      injectReducers(store, combineReducers(require("./createReducer").default))
    );
  }

  return store;
};
