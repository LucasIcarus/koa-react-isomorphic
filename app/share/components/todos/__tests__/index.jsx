import { List } from "immutable";
import React from "react";
import { render } from "enzyme";
import { Provider } from "react-redux";
import createStore from "../../../createStore";
import todosReducer, {
  mountPoint as todosMountPoint,
  setTodos
} from "../logicBundle";
import injectReducers from "../../../helpers/injectReducers";
import Todos from "..";

describe("Component: Todos", () => {
  let store;

  beforeEach(() => {
    store = createStore();
    injectReducers(store, { [todosMountPoint]: todosReducer });
    store.dispatch(
      setTodos(
        List([
          { text: "Todo 1", complete: false },
          { text: "Todo 2", complete: false },
          { text: "Todo 3", complete: false },
          { text: "Todo 4", complete: false }
        ])
      )
    );
  });

  it("should render component", () => {
    expect(
      render(
        <Provider key="provider" store={store}>
          <div>
            <Todos />
          </div>
        </Provider>
      )
    ).toMatchSnapshot();
  });
});
