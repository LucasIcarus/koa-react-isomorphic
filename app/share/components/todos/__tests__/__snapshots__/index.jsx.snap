// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Todos should render component 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <Component />
    <TodosAdd
      relay={Object {}}
      viewer={
        Object {
          "numberOfTodos": 10,
          "todos": Object {
            "edges": Array [
              Object {
                "node": Object {
                  "complete": false,
                  "id": 1,
                  "text": "Todo 1",
                },
              },
            ],
          },
        }
      }
    />
    <TodosBody
      viewer={
        Object {
          "numberOfTodos": 10,
          "todos": Object {
            "edges": Array [
              Object {
                "node": Object {
                  "complete": false,
                  "id": 1,
                  "text": "Todo 1",
                },
              },
            ],
          },
        }
      }
    />
    <Component />
  </div>
</div>
`;
